packages/examples/cvat/recording-oracle/src/handlers/process_intermediate_results.py

```diff
@@ -167,56 +128,86 @@ def _parse_gt(self):
-            except TooFewGtError as e:
-                job_results[job_cvat_id] = self.self.UNKNOWN_QUALITY
-                rejected_jobs[job_cvat_id] = e
-                continue
+        job_id_to_quality_report: dict[int, cvat_api.models.QualityReport] = {}
 
-            job_results[job_cvat_id] = job_mean_accuracy
+        for cvat_task_id in cvat_task_ids:
+            task_quality_report = cvat_api.get_task_quality_report(cvat_task_id)
+            task_quality_report_data = cvat_api.get_quality_report_data(task_quality_report.id)
+            task_id_to_quality_report_data[cvat_task_id] = task_quality_report_data
 
-            for gt_sample in gt_dataset:
-                updated_gt_stats.setdefault(gt_sample.id, _UpdatedFailedGtInfo()).occurrences += 1
+            task_val_layout = cvat_api.get_task_validation_layout(cvat_task_id)
+            honeypot_frame_to_real = {
+                f: task_val_layout.honeypot_real_frames[idx]
+                for idx, f in enumerate(task_val_layout.honeypot_frames)
+            }
+            task_id_to_val_layout[cvat_task_id] = task_val_layout
+            task_id_to_honeypots_mapping[cvat_task_id] = honeypot_frame_to_real
+
+            job_id_to_quality_report.update(
+                {
+                    quality_report.job_id: quality_report
+                    for quality_report in cvat_api.get_jobs_quality_reports(task_quality_report.id)
+                }
+            )
+
+        # accepted jobs from the previous epochs are not included
+        for job_meta in meta.jobs:
+            cvat_job_id = job_meta.job_id
+            cvat_task_id = job_meta.task_id
+
+            # assess quality of the job's honeypots
+            task_quality_report_data = task_id_to_quality_report_data[cvat_task_id]
+            task_honeypots = {int(frame) for frame in task_quality_report_data.frame_results}
+            honeypots_mapping = task_id_to_honeypots_mapping[cvat_task_id]
+
+            for honeypot in task_honeypots & set(job_meta.job_frame_range):
+                val_frame = honeypots_mapping[honeypot]
+
+                result = task_quality_report_data.frame_results[str(honeypot)]
+                self._gt_stats.setdefault((cvat_task_id, val_frame), ValidationFrameStats())
+                self._gt_stats[
+                    (cvat_task_id, val_frame)
+                ].accumulated_quality += result.annotations.accuracy
+
+                if result.annotations.accuracy < min_quality:
+                    self._gt_stats[(cvat_task_id, val_frame)].failed_attempts += 1
+                else:
+                    self._gt_stats[(cvat_task_id, val_frame)].accepted_attempts += 1
+
+            # assess job quality
+            job_quality_report = job_id_to_quality_report[cvat_job_id]
+
+            accuracy = job_quality_report.summary.accuracy
+            if isinstance(accuracy, int):
+                assert accuracy == 0
```

Comment:
What does this check mean?

Reply:
It was used to identify situations where there are no valid gt frames in the job, resulting in an undefined quality. Probably it was not the best approach, replaced by `job_quality_report.summary.gt_count`

Reply:
`gt_count` is the number of gt annotations. I think it's better to check it using `validation_layout` instead. Also, I'm not sure accuracy has to be 0 in such cases.

Reply:
> Also, I'm not sure accuracy has to be 0 in such cases.

Are you talking about the value returned by CVAT? In that case, it is.

Reply:
Ok. Actually, could you clarify how we can get this situation now? It was possible before frames could be changed in jobs, but now we always should have all the jobs verifiable, unless we have more than `val_size` images in the pool.

Reply:
Rigth now such a situation is possible: when recording oracle cannot select new N validation frames, we just update disabled frames in the task. That means that related honeypots are marked as deleted in related jobs, right? So it is possible to achieve the situation, when all validation frames are disabled. 

> but now we always should have all the jobs verifiable, while we have at least val_size images in the pool.

What should we do in case when all validation frames contain incorrect annotations?

Reply:
> What should we do in case when all validation frames contain incorrect annotations?

By the current policy, we stop the escrow annotation with the current annotations without checking them. This part is [here](https://github.com/humanprotocol/human-protocol/blob/34bbdf364473573e2af8bc8049ce6a71b7eff540/packages/examples/cvat/recording-oracle/src/handlers/process_intermediate_results.py#L562-L586).

> Rigth now such a situation is possible: when recording oracle cannot select new N validation frames, we just update disabled frames in the task. That means that related honeypots are marked as deleted in related jobs, right? So it is possible to achieve the situation, when all validation frames are disabled.

Discussed in https://github.com/humanprotocol/human-protocol/pull/2720#discussion_r1834014156

Reply:
If it's not possible to select N new validation frames, it means we don't have enough GT to validate any of the jobs, since each CVAT task has the same GT frames. The escrow annotation has to be finished. You stop any interactions with CVAT, no new annotation iterations should happen for any of the escrow jobs.